src/overlays/actors/ovl_En_Tk/z_en_tk.c

```diff
@@ -17,62 +51,972 @@ const ActorInit En_Tk_InitVars =
+        this->h_1E0 = 0;
+        this->action_func = EnTk_Dig_Update;
+    }
+    else
+    {
+        this->actor.speedXZ = EnTk_Step(this, ctxt);
+        EnTk_Orient(this, ctxt);
+        Math_SmoothScaleMaxMinS(&this->head_rot, 0, 6, 1000, 1);
+        EnTk_CheckCurrentSpot(this);
+
+        DECR(this->action_countdown);
+        if (EnTk_CheckFacingPlayer(this) != 0 || this->action_countdown == 0)
+        {
+            EnTk_Stop(this, ctxt);
+            this->action_func = EnTk_Stop_Update;
+        }
+    }
+}
+
+void EnTk_Dig_Update(EnTk *this, GlobalContext *ctxt)
+{
+    Vec3f reward_origin;
+    Vec3f reward_pos;
+    s32 reward_params[] =
+    {
+        0x0000, /* Green rupee */
+        0x0001, /* Blue rupee */
+        0x0002, /* Red rupee */
+        0x0014, /* Purple rupee */
+        0x0006, /* Heart piece */
+    };
+
+    EnTk_Dig_Eff(this);
+
+    if (this->skel_anim.animCurrentFrame == 32.f)
+    {
+        /* NA_SE_EV_DIG_UP - What's gonna come out? */
+        Audio_PlayActorSound2(&this->actor, 0x28C8);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1BDDC.s")
+        this->reward_timer = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1BE94.s")
+        if (this->valid_dig_here == 1)
+        {
+            reward_origin.x = 0.f;
+            reward_origin.y = 0.f;
+            reward_origin.z = -40.f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1BF14.s")
+            Matrix_RotateY(this->actor.shape.rot.y, 0);
+            Matrix_MultVec3f(&reward_origin, &reward_pos);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1BF8C.s")
+            VEC3_ADD(reward_pos, this->actor.posRot.pos)
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1C00C.s")
+            this->current_reward = EnTk_ChooseReward(this);
+            if (this->current_reward == 3)
+            {
+                /*
+                 * Upgrade the purple rupee reward to the heart piece if this
+                 * is the first grand prize dig.
+                 */
+                if ((gSaveContext.item_get_inf[1] & 0x1000) == 0)
+                {
+                    gSaveContext.item_get_inf[1] |= 0x1000;
+                    this->current_reward = 4;
+                }
+            }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1C0D0.s")
+            Item_DropCollectible(ctxt, &reward_pos,
+                                 reward_params[this->current_reward]);
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Tk/func_80B1C154.s")
+    if (this->skel_anim.animCurrentFrame >= 32.f && this->reward_timer == 10)
+    {
+        /* Play a reward sound shortly after digging */
+        if (this->valid_dig_here == 0)
+        {
+            /* NA_SE_SY_ERROR - Bad dig spot */
+            Audio_PlayActorSound2(&this->actor, 0x4806);
```

Comment:
We have a sfx enum in sfx.h, you can put these directly in there

Reply:
Will do.